utils/pretty.py

```diff
@@ -24,13 +24,55 @@ def x(o):
     return x
 
 
+def process_attrs(attrs):
+    """Enables us for more freedom in attribute specification"""
+    if isinstance(attrs, basestring):
+        return [attrs]
+    elif attrs is None:
+        return []
+    elif isinstance(attrs, (list, tuple)):
+        return attrs
+    else:
+        raise TypeError(
+            "pretty_attrs should be tuple, list, str or None, not {}".format(type(attrs).__name__))
+
+
 class Pretty(object):
-    """A mixin that prints repr as <MyClass field1=..., field2=...>. The
-       fields that will be printed should be stored in the class's
-       pretty_attrs attribute (none by default).
+    """A mixin that prints repr as <MyClass field1=..., field2=...>. Provides default constructor
 
+    The fields that will be printed should be stored in the class's pretty_attrs attribute
+    (none by default). It also provides a default constructor for the values specified.
+
+    the ``pretty_attrs`` can be list, tuple, str or None. The ``required_attrs`` is a list of
+    attributes that are required for a successful construction of the object, same structure as
+    pretty_attrs. ``default_attrs`` is a dictionary of default values. Keys must exist in
+    ``pretty_attrs`` for it to work.
     """
-    pretty_attrs = []
+    pretty_attrs = None
+    required_attrs = None
+    default_attrs = {}
+
+    def __init__(self, *args, **kwargs):
+        pretty_attrs = process_attrs(type(self).pretty_attrs)
+        for attr, value in type(self).default_attrs.iteritems():
+            if attr not in pretty_attrs:
+                raise NameError("No such attribute {}!".format(attr))
+            setattr(self, attr, value)
+        for i, arg in enumerate(args):
+            if i >= len(args):
+                raise ValueError("Count of the arguments is wrong!")
+            setattr(self, pretty_attrs[i], arg)
+        for attr, value in kwargs.iteritems():
+            if attr not in pretty_attrs:
+                raise NameError("No such field {}".format(attr))
+            if hasattr(self, attr):
+                raise NameError(
+                    "The {} field was already set to {}!".format(attr, getattr(self, attr)))
+            setattr(self, attr, value)
```

Comment:
> the **kwargs emulate default values args. That is why the check is there. This has nothing to do with passing anything to the instance. It is about behaving like a "transfer-the-values" constructor.

I don't understand what you mean here. `**kwargs` are the keyworg args passed into the instance. They literally have everything to do with things passed into the instance. Default values appear to come from `default_attrs`, and kwargs shouldn't have anything to do with it aside from looking to see if there are new values in kwargs. Anything else in kwargs **does not matter**. 

If the plan for implementers is to throw away the default constructor (thus losing all of the behaviors of having default and required attrs), then this entire class becomes useless. Alternatively, it could just not care about `**kwargs`, and be a perfectly well-behaved subclass.

Ideally, `__init__` should do no more than this:
```python
def __init__(self, *args, **kwargs):
    for attr, default in self.default_attrs.items():
        if not hasattr(self, attr):
            setattr(self, attr, default)

    for attr in self.required_attrs:
        if not hasattr(self, attr):
            raise AttributeError(attr is required)

```

pretty_attrs is only used by `__repr__`, and pretty attrs aren't required.